beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -750,3 +757,371 @@ func (bs *Server) checkSync(ctx context.Context, w http.ResponseWriter) bool {
+	if ok {
+		handleProduceAltairV3(isSSZ, altairBlock, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+	optimistic, err := bs.OptimisticModeFetcher.IsOptimistic(r.Context())
+	if err != nil {
+		http2.HandleError(w, errors.Wrap(err, "Could not determine if the node is a optimistic node").Error(), http.StatusInternalServerError)
+		return
+	}
+	if optimistic {
+		http2.HandleError(w, "The node is currently optimistic and cannot serve validators", http.StatusInternalServerError)
+		return
+	}
+	blindedBellatrixBlock, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_BlindedBellatrix)
+	if ok {
+		handleProduceBlindedBellatrixV3(isSSZ, blindedBellatrixBlock, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+	bellatrixBlock, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_Bellatrix)
+	if ok {
+		handleProduceBellatrixV3(isSSZ, bellatrixBlock, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+	blindedCapellaBlock, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_BlindedCapella)
+	if ok {
+		handleProduceBlindedCapellaV3(isSSZ, blindedCapellaBlock, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+	capellaBlock, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_Capella)
+	if ok {
+		handleProduceCapellaV3(isSSZ, capellaBlock, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+	blindedDenebBlockContents, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_BlindedDeneb)
+	if ok {
+		handleProduceBlindedDenebV3(isSSZ, blindedDenebBlockContents, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+	denebBlockContents, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_Deneb)
+	if ok {
+		handleProduceDenebV3(isSSZ, denebBlockContents, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+}
+
+func handleProducePhase0V3(isSSZ bool, phase0Block *eth.GenericBeaconBlock_Phase0, w http.ResponseWriter, validate *validator.Validate, isBlinded bool, payloadValue uint64) {
+	w.Header().Set(api.ExecutionPayloadBlindedHeader, fmt.Sprintf("%v", isBlinded))
+	w.Header().Set(api.ExecutionPayloadValueHeader, fmt.Sprintf("%d", payloadValue))
+	if isSSZ {
+		sszResp, err := phase0Block.Phase0.MarshalSSZ()
+		if err != nil {
+			http2.HandleError(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		http2.WriteSsz(w, sszResp, "phase0Block.ssz")
+		return
+	}
+	block, err := convertInternalBeaconBlock(phase0Block.Phase0)
+	if err != nil {
+		http2.HandleError(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if err = validate.Struct(block); err == nil {
+		http2.WriteJson(w, &Phase0ProduceBlockV3Response{
+			Version:                 version.String(version.Phase0),
+			ExecutionPayloadBlinded: false,
+			ExeuctionPayloadValue:   fmt.Sprintf("%d", payloadValue), // mev not available at this point
```

Comment:
what does this comment mean?

Reply:
I was thinking since it's local build it's not available 🤔 but maybe i'm wrong in that assumption